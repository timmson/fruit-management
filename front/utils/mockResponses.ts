export function okAndJson(json: any): Promise<Response> {
    return Promise.resolve({
        arrayBuffer(): Promise<ArrayBuffer> {
            return Promise.resolve(undefined);
        },
        blob(): Promise<Blob> {
            return Promise.resolve(undefined);
        },
        body: undefined,
        bodyUsed: false,
        clone(): Response {
            return undefined;
        },
        formData(): Promise<FormData> {
            return Promise.resolve(undefined);
        },
        headers: undefined,
        json(): Promise<any> {
            return Promise.resolve(json);
        },
        ok: true,
        redirected: false,
        status: 200,
        statusText: "OK",
        text(): Promise<string> {
            return Promise.resolve("");
        },
        type: undefined,
        url: ""
    })
}