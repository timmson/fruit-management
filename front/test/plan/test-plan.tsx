import React from "react"
import renderer from "react-test-renderer"
import Context from "../../src/context"
import {okAndJson} from "../../utils/mockResponses"
import Plan from "../../src/plan/plan"

global.fetch = jest.fn(() => okAndJson({}))

describe("Plan should", () => {

	test("return plan form", () => {
		const context = []

		const component = renderer.create(
			<Context.Provider value={context}>
				<Plan/>
			</Context.Provider>
		)

		expect(component.toJSON()).toMatchSnapshot()

		component.unmount()
	})

})