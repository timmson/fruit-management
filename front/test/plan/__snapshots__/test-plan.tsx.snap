// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan should return plan form 1`] = `
[
  "Релизы: ",
  <br />,
  <div
    style={
      {
        "float": "right",
      }
    }
  >
    <a
      href="?section=agile"
    >
      Гибкое планирование
    </a>
  </div>,
  <br />,
  <h3>
    Календарное планирование
  </h3>,
  <table
    className="basic w-100"
  >
    <thead>
      <tr>
        <th
          rowSpan="2"
        >
          Пользователь
        </th>
        <th
          rowSpan="2"
        >
          Длина
        </th>
        <th
          colSpan={0}
        >
          7
        </th>
      </tr>
      <tr />
    </thead>
    <tbody />
  </table>,
  <br />,
  <h3>
    Структура
  </h3>,
  <table
    className="basic w-100"
  >
    <thead>
      <tr>
        <th
          style={
            {
              "width": "15%",
            }
          }
        >
          Запрос
        </th>
        <th
          style={
            {
              "width": "40%",
            }
          }
        >
          Задача
        </th>
        <th>
          Исполнитель
        </th>
        <th>
          %
        </th>
        <th>
          Приоритет
        </th>
        <th>
          Статус
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
